parsimony/functions/penalties.py

```diff
@@ -1251,13 +1260,19 @@ def lambda_max(self):
         # should allow dense matrices as well.
         if self._lambda_max is None:
 
-            from parsimony.algorithms.nipals import RankOneSparseSVD
-
-            A = self.A
-            # TODO: Add max_iter here!
-            v = RankOneSparseSVD().run(A)  # , max_iter=max_iter)
-            us = A.dot(v)
-            self._lambda_max = np.sum(us ** 2.0)
+            if self.A != None:
+                from parsimony.algorithms.nipals import RankOneSparseSVD
+                A = self.A
+                # TODO: Add max_iter here!
+                v = RankOneSparseSVD().run(A)  # , max_iter=max_iter)
+                us = A.dot(v)
+                self._lambda_max = np.sum(us ** 2.0)
```

Comment:
> Can you double check this?

Here it is squared

Reply:
Right, this would be the eigenvalue of A, the squared norm of the score vectors in a PCA.